app/models/submission.rb

```diff
@@ -248,6 +250,20 @@ def new_version_needed?
       )
     }
     can :view_turnitin_report
+
+    given { |user, session|
```

Comment:
This seems like it should be generic.  Maybe :view_plagiarism_report with combined logic and/or helper functions.